[Improve] Login Screen UI Update

aliseyalvi opened the pull request:
### What does this PR?
Updated login sceen UI according to ecency web login UI

### Where should the reviewer start?
Go to Login Screen on mobile app and check updated UI

### Issue number
fixes #2793 

### Screenshots/Video
<img width="335" alt="image" src="https://github.com/ecency/ecency-mobile/assets/48380998/57b992e9-e62e-4f1d-8d48-4aedaf75b3e7">

aliseyalvi commented:
<img width="317" alt="image" src="https://github.com/ecency/ecency-mobile/assets/48380998/73f73567-6da2-4ce6-a8c3-8e6f03cd3c16">

<img width="404" alt="image" src="https://github.com/ecency/ecency-mobile/assets/48380998/f6140fe2-2fec-4510-9c5b-a7ccd9373a20">

aliseyalvi commented:
@noumantahir fixed [this issue](https://discord.com/channels/@me/920267778190086205/1178935396692934657) as well

noumantahir commented on src/components/orDivider/orDividerView.tsx:
```diff
@@ -0,0 +1,18 @@
+  return (
+    <View style={[styles.dividerContainer, containerStyle]}>
+      <View style={[styles.divider, styles.leftDivider]} />
+      <Text style={styles.orText}>OR</Text>
```
use intl

noumantahir reviewed (changes requested):
Check how the input skips characters some times... this has been there for quite some time now... if we are improving Login, I think this can be addressed as well...

https://github.com/ecency/ecency-mobile/assets/6298342/fe90d266-34e8-4e55-b923-2ff69b6e943b

Do also check one comment in PR thread. Other than that I think we can wrap this up and move to signup screen next.

aliseyalvi commented:
@noumantahir 

- converted login screen class component to functional component and hooks
- updated debounce logic which was causing characters glitch
- used intl

https://github.com/ecency/ecency-mobile/assets/48380998/c443a1d5-afdb-4f60-9faf-56f15655e7a1

noumantahir reviewed (approved):
lgtm

feruzm reviewed (approved):
lgtm, would be better if we converted those modified components/pages to typescript.
